test/skimNtuple_HHbtag.cpp

```diff
@@ -5271,29 +5318,52 @@ int main (int argc, char** argv)
 		  theSmallTree.m_HHbregrsvfit_m	  = tlv_HHbregrsvfit.M();
 
-		  // Getting ParticleNet scale factors from the pnetSF.cc
-		  pnetSF pnetSF_helper;
-		  std::map<std::string, std::vector<float>> pnetSF_map = pnetSF_helper.getSFmap(tlv_fj.Pt(), PERIOD);
-
-		  if (isHHsignal){
-		    // saving each working point
-		    std::tuple<float, float, float> SF_HP = pnetSF_helper.getSF(pnetSF_map, "HP");
-		    theSmallTree.m_fatjet_particleNetMDJetTags_HP_SF      = std::get<0>(SF_HP);
-		    theSmallTree.m_fatjet_particleNetMDJetTags_HP_SF_up   = std::get<1>(SF_HP);
-		    theSmallTree.m_fatjet_particleNetMDJetTags_HP_SF_down = std::get<2>(SF_HP);
-
-		    std::tuple<float, float, float> SF_MP = pnetSF_helper.getSF(pnetSF_map, "MP");
-		    theSmallTree.m_fatjet_particleNetMDJetTags_MP_SF      = std::get<0>(SF_MP);
-		    theSmallTree.m_fatjet_particleNetMDJetTags_MP_SF_up   = std::get<1>(SF_MP);
-		    theSmallTree.m_fatjet_particleNetMDJetTags_MP_SF_down = std::get<2>(SF_MP);
-
-		    std::tuple<float, float, float> SF_LP = pnetSF_helper.getSF(pnetSF_map, "LP");
-		    theSmallTree.m_fatjet_particleNetMDJetTags_LP_SF      = std::get<0>(SF_LP);
-		    theSmallTree.m_fatjet_particleNetMDJetTags_LP_SF_up   = std::get<1>(SF_LP);
-		    theSmallTree.m_fatjet_particleNetMDJetTags_LP_SF_down = std::get<2>(SF_LP);
+		  // verify if the fatjet matches any previously found Higgs or Z decaying into bb
+		  bool matchedToHiggs = false;
+		  bool matchedToZ = false;
+		  for (const auto& higgs : higgsVectors) {
+		    if (tlv_fj.DeltaR(higgs) < 0.8) {
+		      matchedToHiggs = true;
+		      break;
+		    }
+		  }
+		  for (const auto& zboson : zBosonVectors) {
+		    if (tlv_fj.DeltaR(zboson) < 0.8) {
+		      matchedToZ = true;
+		      break;
+		    }
+		  }
+
+		  theSmallTree.m_fatjet_isMatchedToHiggs = matchedToHiggs;
+		  theSmallTree.m_fatjet_isMatchedToZ = matchedToZ;
+
+		  // determine sample type
+		  std::string sample_type = "";
+		  if (matchedToHiggs || matchedToZ){
+		    sample_type = "HHlike";
+		  }
+		  else if  (isDYlike){
+		    sample_type = "DYlike";
+		  }
+		  else if (isTTlike){
+		    sample_type = "TTlike";
 		  }
-		  //else 
-		  //     --> handle bkg samples differently . For now, their values remain set to 1
+		  
+		  // Getting ParticleNet scale factors from the class
+		  if (sample_type != ""){
+		    auto PNetAK8SF = PNetSFInterface(PERIOD);
+		    std::vector<float> PNetSFmap = PNetAK8SF.getSFvec(tlv_fj.Pt(), sample_type);
+		    theSmallTree.m_fatjet_particleNetMDJetTags_HP_SF      = PNetSFmap[0];
+		    theSmallTree.m_fatjet_particleNetMDJetTags_HP_SF_up   = PNetSFmap[1];
+		    theSmallTree.m_fatjet_particleNetMDJetTags_HP_SF_down = PNetSFmap[2];
+		    theSmallTree.m_fatjet_particleNetMDJetTags_MP_SF      = PNetSFmap[3];
+		    theSmallTree.m_fatjet_particleNetMDJetTags_MP_SF_up   = PNetSFmap[4];
+		    theSmallTree.m_fatjet_particleNetMDJetTags_MP_SF_down = PNetSFmap[5];
```

Comment:
Is is worth to store these? We are only using the LP and want to reduce the size of skims whenever possible.